### PR TITLE
Make fields more flat

### PR DESCRIPTION
**Pull request by akatsoulas**
* Replace categories with keyword.

### CODE REVIEW

**Comment by LeoMcA** [comment]
As an abstraction, I think this is good - but I'm not sure if we want the abstraction. My thinking has been we probably want to move everything towards the implementation used in `QuestionDocument` which doesn't use any `InnerDoc`s, and just uses (lists of) ids. We don't need the title of a product or group to filter on them or aggregate them, their SQL id is enough.

However, storing titles for products would be useful if we want to use product titles (and even descriptions) in search, and store them properly in a `SumoLocaleAwareTextField` as well. So if we matched across `content.en-US` and `product.title.en-US` then a search for "enable do not track in firefox for android" (without any filtering) could cause an answer like "here's how to enable do not track..." in the Firefox for Android AAQ to match higher than "here's how to enable do not track in firefox..." in the Firefox AAQ.

But then again, doing this means we have to reindex everything if a product name changes, or a localisation of a product name changes, and it might be better to detect in python if someone has placed a product name in their query and then suggest they filter by that, or boost the scores of documents with that product's id instead.

**Comment by akatsoulas** [comment]
@LeoMcA I updated the PR to change a few fields and make others more flat by using only IDs.

File: kitsune/search/v2/documents.py [hunk 1]
```diff
@@ -129,7 +137,7 @@ class Index:
         using = config.DEFAULT_ES7_CONNECTION
 
     def prepare_question_tag_id(self, instance):
```
Comment (LeoMcA) [inline kitsune/search/v2/documents.py:139]: I'd also rename the field to `question_tag_ids` to be consistent with our other naming

File: kitsune/search/v2/documents.py [hunk 2]
```diff
@@ -272,11 +252,11 @@ def prepare_avatar(self, instance):
...
 
-    def prepare_products(self, instance):
-        return [ProductInnerDoc.prepare(product) for product in instance.products.all()]
+    def prepare_product_ids(self, instance):
+        return [product.id for product in instance.products.all()]
 
-    def prepare_groups(self, instance):
-        return [GroupInnerDoc.prepare(groups) for groups in instance.user.groups.all()]
+    def prepare_group_ids(self, instance):
+        return [group.id for group in instance.groups.all()]
```
Comment (LeoMcA) [inline kitsune/search/v2/documents.py:259]: This needs to be `instance.user.groups.all()`

**Review by LeoMcA — APPROVED** [review]
r+wc

**Pull request merged**